xapian-bindings/python/doxy2swig.py

```diff
@@ -39,7 +40,7 @@ def my_open_write(dest):
     if hasattr(dest, "write"):
         return dest
     else:
-        return open(dest, 'w')
+        return io.open(dest, 'w', encoding = 'utf8')
 
 
 class Doxy2SWIG:
```

Comment:
This is a useful change, but not really related to the stated purpose of the PR ("Add support for BM25+ weighting function") - it just happens this change adds the first instance of a non-ASCII character to the C++ documentation comments (or at least to those which are taken through to Python).  So it would arguably be better to deal with separately.

It's also incomplete - the same issue affects Python3.

And for Python2, even with the fix in this commit I get an error when the Python interpreter tries to import the wrapper due to the default encoding for Python2 being ASCII (since 2.5 - before that it was latin-1 which wouldn't give an error, but wouldn't work as intended):

```
/usr/bin/python2 -c 'import xapian'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "xapian/__init__.py", line 3099
SyntaxError: Non-ASCII character '\xc3' in file xapian/__init__.py on line 3100, but no encoding declared; see http://python.org/dev/peps/pep-0263/ for details
```

Python3's default source encoding if UTF-8, so no equivalent change is needed.

So I've cherry-picked this commit as 215c8cce2df208a0b3cea3ed5f2ec410b340de90, with the identical fix applied for Python3, and then committed c277f4422cfa15aeefa4c7692c16f3763c0af68e to specify the source ending for Python2.